### PR TITLE
FUSETOOLS2-1097 - provide completion for kamelet id

### PR DESCRIPTION
**Pull request by apupier**
first iteration:
- ~requires locally built version until version is published on Apache
Snapshots. the version is currently main-SNAPSHOT, this will need to be updated.~ https://repository.apache.org/content/repositories/snapshots/org/apache/camel/kamelets/camel-kamelets-catalog/main-SNAPSHOT/
- no distinction between source and sink
- no description
- Catalog is reloaded on each completion call in a kamelet template Id
part

Screenshot from 2021-05-04 09-30-56

### CODE REVIEW

**Comment by apupier** [comment]
no release of kamelet catalog planned soon. i think we can go with snpashot waiting for it (will most probably happen after Productized Camel K)

**Review by lhein — APPROVED** [review]

File: src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PathParamURIInstance.java [hunk 1]
```diff
@@ -69,8 +70,11 @@ public String getValue() {
 	@Override
 	public CompletableFuture<List<CompletionItem>> getCompletions(CompletableFuture<CamelCatalog> camelCatalog, int positionInCamelUri, TextDocumentItem docItem, SettingsManager settingsManager) {
 		if(pathParamIndex == 0) {
-			if("kafka".equals(uriInstance.getComponentName())) {
+			String componentName = uriInstance.getComponentName();
+			if("kafka".equals(componentName)) {
```
Comment (bfitzpat) [inline src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PathParamURIInstance.java:74]: Wondering if "kafka" and "kamelet" should be constants somewhere?

**Review by bfitzpat — APPROVED** [review]
One minor question, but looks like a good first step!

File: src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PathParamURIInstance.java (hunk 1 above)
Comment (apupier) [reply src/main/java/com/github/cameltooling/lsp/internal/instancemodel/PathParamURIInstance.java:74]: I was also wondering it :-)

not sure where to put them:
- in this class, as currently used only here (but they will surely be used elsewhere soon)
- in a new specific class ComponentNames
- Each in a specific kafka/Kamelet class. a new one or reusing a "CompletionProvider" one

**Pull request merged**